e2e/commands_test.go

```diff
@@ -265,3 +268,107 @@ func TestAttachmentsWithRegistry(t *testing.T) {
 	assert.Assert(t, strings.Contains(resultOutput, "nesteddir/config2.cfg"))
 	assert.Assert(t, strings.Contains(resultOutput, "nesteddir/nested2/nested3/config3.cfg"))
 }
+
+const (
+	targetName       = "testTargetContext"
+	credsName        = "creds-test"
+	installationName = "test-simple-installation"
+)
+
+func TestBundle(t *testing.T) {
+	dir := fs.NewDir(t, "test-bundle")
+	defer dir.Remove()
+
+	outputPath := filepath.Join(dir.Path(), "test-bundle.json")
+	simpleDockerApp := filepath.Join("testdata", "simple", "simple.dockerapp")
+	icmd.RunCmd(icmd.Cmd{
+		Command: []string{dockerApp, "bundle", "-o", outputPath},
+		Dir:     simpleDockerApp,
+	})
+
+	b, err := ioutil.ReadFile(outputPath)
+	assert.NilError(t, err)
+	assert.Assert(t, golden.String(string(b), "simple-bundle.golden"))
+}
+
+func TestLifeCycle(t *testing.T) {
+	simpleDockerApp := filepath.Join("testdata", "echo.dockerapp")
+
+	// Install
+	icmd.RunCommand(dockerApp, "install", simpleDockerApp,
+		"--name", installationName,
+		"--target-context", targetName).
+		Assert(t, icmd.Success)
+
+	// Double install must fail
+	icmd.RunCommand(dockerApp, "install", simpleDockerApp,
+		"--name", installationName,
+		"--target-context", targetName).
+		Assert(t, icmd.Expected{ExitCode: 1})
+
+	// Status for 1 replica
+	assert.NilError(t, waitUntilPortIsOpen(8080, time.Now().Add(10*time.Second)))
+	icmd.RunCommand(dockerApp, "status", installationName,
+		"--target-context", targetName).
+		Assert(t, icmd.Success)
+	assert.NilError(t, checkStatusUntilDeadline(t, "1/1", time.Now().Add(10*time.Second)))
+
+	// Upgrade
+	icmd.RunCommand(dockerApp, "upgrade", installationName,
+		"--set", "echoapp.replicas=2",
+		"--target-context", targetName).
+		Assert(t, icmd.Success)
+
+	// Status for 2 replicas
+	icmd.RunCommand(dockerApp, "status", installationName,
+		"--target-context", targetName).
+		Assert(t, icmd.Success)
+	assert.NilError(t, checkStatusUntilDeadline(t, "2/2", time.Now().Add(10*time.Second)))
```

Comment:
golden that too?

Reply:
That would be great, but `status` includes the `ID` which is dynamic :(